mesonbuild/modules/windows.py

```diff
@@ -51,9 +51,15 @@ def compile_resources(self, state, args, kwargs):
             for arg in extra_args:
                 if ' ' in arg:
                     mlog.warning(m.format(arg))
-            # Pick-up env var WINDRES if set. This is often used for specifying
-            # an arch-specific windres.
-            rescomp_name = os.environ.get('WINDRES', 'windres')
+            rescomp_name = None
+            if state.environment.is_cross_build():
```

Comment:
Normally you'd have to account for the [`native: true`](https://github.com/mesonbuild/meson/wiki/Reference%20manual#executable) use-case (building an exe for use during the build, which means it must be built to run on the build machine), and in this case adding that kwarg to `compile_resources()` sucks. We need something like #1531.

However, you'd only encounter this if you were building for ARM on a Windows machine and somehow your build tool also needed to use resources, which is an extreme edge-case. So we should ignore that case and handle it in https://github.com/mesonbuild/meson/issues/1531. However, we *should* add a note here like:

```
# FIXME: Does not handle `native: true` executables, see
# https://github.com/mesonbuild/meson/issues/1531
```